JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt

```diff
@@ -131,20 +137,45 @@ fun ArticleScreen(
         topBar = {
             InsetAwareTopAppBar(
                 title = {
-                    Text(
-                        text = "Published in: ${post.publication?.name}",
-                        style = MaterialTheme.typography.subtitle2,
-                        color = LocalContentColor.current
-                    )
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .wrapContentWidth(align = Alignment.CenterHorizontally)
+                            .padding(start = 30.dp)
+                    ) {
+                        Surface(
+                            color = Color(0xff073042),
+                            shape = CircleShape,
+                            modifier = Modifier.size(36.dp, 36.dp)
+                        ) {
+                            Image(
+                                painter = painterResource(R.drawable.ic_article_icon),
+                                contentDescription = "",
+                                colorFilter = ColorFilter.tint(Color(0xff32dd84)),
+                                modifier = Modifier.padding(4.dp)
+                            )
+                        }
+                        Text(
+                            text = stringResource(id = R.string.published_in, post.publication?.name ?: ""),
+                            style = MaterialTheme.typography.subtitle2,
+                            color = LocalContentColor.current,
+                            modifier = Modifier
+                                .padding(start = 10.dp)
+                                .weight(1.5f)
+                        )
+                    }
                 },
                 navigationIcon = {
                     IconButton(onClick = onBack) {
-                        Icon(
+                        Image(
                             imageVector = Icons.Filled.ArrowBack,
-                            contentDescription = stringResource(R.string.cd_navigate_up)
+                            contentDescription = stringResource(R.string.cd_navigate_up),
+                            colorFilter = ColorFilter.tint(MaterialTheme.colors.primary)
                         )
                     }
-                }
+                },
+                elevation = 0.dp,
```

Comment:
Setting this Elevation to 0dp, what happens when you start scrolling? Do you see the text of the screen content cut off? If so, that's not the expected behavior. I'd suggest sticking with the elevation, or better yet, adding elevation on scroll.

Reply:
The behavior is the same when scrolling with 0.dp elevation, the reason why I remove the elevation is because it sets a shadow under the top bar that doesn't match the design


**App with default 4.dp elevation**
![image](https://user-images.githubusercontent.com/16503982/118527083-87be1400-b706-11eb-9d8a-1289ad6ef17b.png)

**Figma design**
![image](https://user-images.githubusercontent.com/16503982/118527009-72e18080-b706-11eb-9f57-62fcf3f34ff4.png)

Reply:
What I mean: If there's a shadow showing, the text that scrolls out of view is not cut off.

With shadow vs without shadow
<img src="https://user-images.githubusercontent.com/6952116/118800870-ace68a00-b8a0-11eb-9568-7bb4a5981212.png" width="200px"/> <img src="https://user-images.githubusercontent.com/6952116/118800880-aeb04d80-b8a0-11eb-805c-b39176fefefb.png" width="200px"/>

Ideally, we'd give the top app bar an elevation when the user scrolls through the content, but hide the elevation when the offset is at 0 dp. But I can imagine that's too much, so just adding the elevation at all times would be an option. Maybe something to discuss with a designer?

Reply:
Got it, I just added the elevation only when offset is bigger than 0

![image](https://user-images.githubusercontent.com/16503982/118883844-576ba680-b8bb-11eb-9048-16f2a126d972.png)
![image](https://user-images.githubusercontent.com/16503982/118883870-605c7800-b8bb-11eb-9e79-1598f9ecd2a2.png)